jenkins_agent_k8s_rock/rockcraft.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: jenkins-agent-k8s
+summary: Jenkins-agent-k8s rock
+description: Jenkins-agent-k8s OCI image for the Jenkins-agent-k8s charm
+version: "1.0"
+base: ubuntu:22.04
+build-base: ubuntu:22.04
+license: Apache-2.0
+platforms:
+  amd64:
+parts:
+  add-user:
+    plugin: nil
+    overlay-script: |
+      mkdir -p $CRAFT_OVERLAY/var/{lib/jenkins,lib/jenkins/agents,log/jenkins}
+      chmod 755 $CRAFT_OVERLAY/var/{lib/jenkins,lib/jenkins/agents,log/jenkins}
+      groupadd -R $CRAFT_OVERLAY --gid 2000 jenkins
+      useradd -R $CRAFT_OVERLAY --system --gid 2000 --uid 2000 --home /var/lib/jenkins jenkins
+  jenkins:
+    plugin: nil
+    overlay-packages:
+      - bash
+      - ca-certificates-java
+      - default-jre-headless
```

Comment:
The server and agent java versions currently don't match using `openjdk-8-jre-headless` chisel, perhaps this could be tracked as another story for future :D 
```
 Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: hudson/remoting/Launcher has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Reply:
I'm not sure if my approach wasn't quite correct, but it seems that chisel cannot be used in rockcraft. If they are used as a build base, `'bare', 'ubuntu:18.04', 'ubuntu:20.04', 'ubuntu:22.04' in field 'base'` are the allowed values.
If used as stage-package, `Detailed information: :: error: invalid slice reference: "ubuntu/jre:17_edge"` error occurs. I'll keep on eye on when this does get released though! Thank you for the reference!